packages/changeset/src/helpers/validate-changeset.ts

```diff
@@ -11,48 +11,45 @@ import { assert } from '@ember/debug';
  * - pass this helper into the form's `@validate` hook `@validate={{validateChangeset}}`
  * - opt-in to `@dataMode="mutable"`
  */
-const validateChangeset: FormValidateCallback<EmberChangeset> = async (
-  changeset,
-  fields
-) => {
-  assert(
-    'Cannot use `validateChangeset` on `@data` that is not a Changeset instance!',
-    isChangeset(changeset)
-  );
-
-  // there is also an argument-less version of changeset.validate(), but for this to work the changeset needs a so called validationMap, and not just a validator function
-  // while ember-changeset-validations would provide such a map, we cannot necessarily rely on it being present, so the way to reliably validate all fields is to iterate
-  // over them explicitly
-  //
-  await Promise.all(fields.map((field) => changeset.validate(field)));
-
-  if (changeset.get('isValid')) {
-    return;
-  }
-
-  const errorRecord: ErrorRecord<Record<string, unknown>> = {};
-
-  for (const { key, value, validation } of changeset.get('errors')) {
-    if (!errorRecord[key]) {
-      errorRecord[key] = [];
+export default function validateChangeset(): FormValidateCallback<EmberChangeset> {
+  return async (changeset, fields) => {
```

Comment:
The diff here makes this change look more complex than it is: basically the helper is not the validation function itself anymore, but a function that _returns_ that validation function. Nothing else has changed here.